### PR TITLE
T/53: Fix mention UI when moving selection inside mention.

### LINKED ISSUE #53: The mention ui opens up when moving selection inside mention
The bug:
Selection_300

Behavior:
Peek 2019-04-10 16-03

This occurs only when selection moves from right to left (if it is between last and previous character).

The selection is properly post-fixed when on boundary.

As visible in the inspector the selection has `mention` attribute but in check of mention UI it is undefined because it is called **after** the mention attribute is re-set to selection.

### PR DESCRIPTION
**Pull request by jodator**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Mention UI stays closed when moving selection inside mention. Closes #53.

---

### CODE REVIEW

File: tests/mentionui.js [hunk 1]
```diff
@@ -560,6 +560,43 @@ describe( 'MentionUI', () => {
...
+
+				model.change( writer => {
+					writer.setAttribute( 'mention', { id: '@Lily', _uid: 1234 }, doc.selection.getFirstRange() );
+				} );
+
+				return waitForDebounce()
+					.then( () => {
+						expect( panelView.isVisible ).to.be.false;
+
+						model.change( writer => {
```
Comment (mlewand) [inline tests/mentionui.js]: This particular part doesn't bring any value to the tests as it puts the selection `foo @Lily ^bar`, which obviously should not trigger a selection but is tested elsewhere. So this step should be skipped.

The test should put following selections:
* `foo @Lily^ bar`
* `foo @Lil^y bar`

Comment (mlewand) [reply tests/mentionui.js]: Fixed in 7879d5c.

Comment (jodator) [reply tests/mentionui.js]: I don'tr recall that case ATM but it might be issue with moving selection from right to left in that specificy scenario.

Comment (mlewand) [reply tests/mentionui.js]: It couldn't be anything like that, I have double checked it against `src/mentionui.js` `src/textwatcher.js` at (former) `master` branch and these assertions obviously didn't fail.

The simplified assertion does the job of detecting the former issue with less steps, and timeouts.

**Pull request merged**